src/finalizer/utils/linea/common.ts

```diff
@@ -96,6 +108,80 @@ export function makeGetMessagesWithStatusByTxHash(
   };
 }
 
+// Temporary re-implementation of the SDK's `L1ClaimingService.getMessageStatus` functions that allow us to use
+// our custom provider, with retry and caching logic, to get around the SDK's hardcoded logic to query events
+// from 0 to "latest" which will not work on all RPC's.
+async function getL2L1MessageStatusUsingCustomProvider(
+  messageService: L1ClaimingService,
+  messageHash: string,
+  l1Provider: ethers.providers.Provider,
+  l1SearchConfig: EventSearchConfig,
+  l2Provider: ethers.providers.Provider,
+  l2SearchConfig: EventSearchConfig
+): Promise<OnChainMessageStatus> {
+  const l2Contract = getL2MessageServiceContractFromL1ClaimingService(messageService, l2Provider);
+  const messageEvent = await getMessageSentEventForMessageHash(messageHash, l2Contract, l2SearchConfig);
+  const l1Contract = getL1MessageServiceContractFromL1ClaimingService(messageService, l1Provider);
+  const [l2MessagingBlockAnchoredEvents, isMessageClaimed] = await Promise.all([
+    getL2MessagingBlockAnchoredFromMessageSentEvent(messageEvent, l1Contract, l1SearchConfig),
+    l1Contract.isMessageClaimed(messageEvent.args?._nonce),
+  ]);
+  if (isMessageClaimed) {
+    return OnChainMessageStatus.CLAIMED;
+  }
+  if (l2MessagingBlockAnchoredEvents.length > 0) {
+    return OnChainMessageStatus.CLAIMABLE;
+  }
+  return OnChainMessageStatus.UNKNOWN;
+}
+export function getL2MessageServiceContractFromL1ClaimingService(
+  l1ClaimingService: L1ClaimingService,
+  l2Provider: ethers.providers.Provider
+): Contract {
+  return new Contract(
+    l1ClaimingService.l2Contract.contract.address,
+    l1ClaimingService.l2Contract.contract.interface,
+    l2Provider
+  );
```

Comment:
Could this alternatively work?

```suggestion
  return l1ClaimingService.l2Contract.contract.connect(l2Provider);
```

Reply:
5a2ad1e0048dc00ba96daa520490ce407b977ef0